<a href="https://colab.research.google.com/github/qjpbtuzara-ai/CPE311-CPE22S3/blob/main/Seatwork7_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Name: Tuzara, John Paul B. <br>
Course and Section: CPE311-CPE22S3 <br>
Date: 2/12/26 <br>
Instructor: Engr. Neal Barton James Matira

Using the meteorite data from the Meteorite_Landings.csv file, update the year column to only contain the year, convert it to a numeric data type, and create a new column indicating whether the meteorite was observed falling before 1970. Set the index to the id column and extract all the rows with IDs between 10,036 and 10,040 (inclusive) with loc[].

    Hint 1: Use year.str.slice() to grab a substring.
    Hint 2: Make sure to sort the index before using loc[] to select the range.

Bonus: There's a data entry error in the year column. Can you find it? (Don't spend too much time on this.)

In [18]:
import pandas as pd

#1. Load csv file
df = pd.read_csv('Meteorite_Landings.csv')

print(df.head())
print(df.info())
print(df['year'].unique()[:10])

       name   id nametype     recclass  mass (g)  fall  \
0    Aachen    1    Valid           L5      21.0  Fell   
1    Aarhus    2    Valid           H6     720.0  Fell   
2      Abee    6    Valid          EH4  107000.0  Fell   
3  Acapulco   10    Valid  Acapulcoite    1914.0  Fell   
4   Achiras  370    Valid           L6     780.0  Fell   

                     year    reclat    reclong           GeoLocation  
0  01/01/1880 12:00:00 AM  50.77500    6.08333     (50.775, 6.08333)  
1  01/01/1951 12:00:00 AM  56.18333   10.23333  (56.18333, 10.23333)  
2  01/01/1952 12:00:00 AM  54.21667 -113.00000    (54.21667, -113.0)  
3  01/01/1976 12:00:00 AM  16.88333  -99.90000     (16.88333, -99.9)  
4  01/01/1902 12:00:00 AM -33.16667  -64.95000   (-33.16667, -64.95)  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45716 entries, 0 to 45715
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   name         45716 non-null

In [16]:

#2. Update year column via slice
df['year'] = df['year'].str.slice(6, 10)

#3. Convert to numeric
df['year'] = pd.to_numeric(df['year'], errors='coerce')

#4. Create new column
df['observed_before_1970'] = df['year'] < 1970

#5. Set index to 'id' and sort
df = df.set_index('id').sort_index()

#6. Extract rows between 10,036 and 10,040 (inclusive with loc)
subset = df.loc[10036:10040]

print("Subset of IDs 10036 to 10040:")
print(subset)

#Bonus: Find error
print("\nPotential errors in year:")
print(df[df['year'] > 2024]['year'])
print(df[df['year'] < 860]['year'])

Subset of IDs 10036 to 10040:
            name nametype         recclass  mass (g)   fall    year    reclat  \
id                                                                              
10036     Enigma    Valid               H4      94.0  Found  1967.0  31.33333   
10037       Enon    Valid  Iron, ungrouped     763.0  Found  1883.0  39.86667   
10038      Enshi    Valid               H5    8000.0   Fell  1974.0  30.30000   
10039  Ensisheim    Valid              LL6  127000.0   Fell  1491.0  47.86667   

         reclong            GeoLocation  observed_before_1970  
id                                                             
10036  -82.31667  (31.33333, -82.31667)                  True  
10037  -83.95000     (39.86667, -83.95)                  True  
10038  109.50000          (30.3, 109.5)                 False  
10039    7.35000       (47.86667, 7.35)                  True  

Potential errors in year:
id
57150    2101.0
Name: year, dtype: float64
Series([], Name: year, dty

In [17]:
print(10040 in df.index)
print(df.index.max())

#ID is skipped

True
57458


The error is that there is a meteor landing in which has not occured yet in year 2101.